examples/air-quality-sensor-app/cc32xx/main/AppTask.cpp

```diff
@@ -0,0 +1,217 @@
+using namespace ::chip::DeviceLayer;
+using namespace ::chip::DeviceManager;
+
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+static TaskHandle_t sAppTaskHandle;
+static QueueHandle_t sAppEventQueue;
+
+AppTask AppTask::sAppTask;
+
+static DeviceCallbacks EchoCallbacks;
+
+int AppTask::StartAppTask()
+{
+    int ret = 0;
+
+    sAppEventQueue = xQueueCreate(APP_EVENT_QUEUE_SIZE, sizeof(AppEvent));
+    if (sAppEventQueue == NULL)
+    {
+        PLAT_LOG("Failed to allocate app event queue");
+        while (true)
+            ;
+    }
+
+    // Start App task.
+    if (xTaskCreate(AppTaskMain, "APP", APP_TASK_STACK_SIZE / sizeof(StackType_t), NULL, APP_TASK_PRIORITY, &sAppTaskHandle) !=
+        pdPASS)
+    {
+        PLAT_LOG("Failed to create app task");
+        while (true)
+            ;
+    }
+
+    // Start App task.
+    if (xTaskCreate(AirQualityTaskMain, "AIR-APP", APP_TASK_STACK_SIZE / sizeof(StackType_t), NULL, APP_TASK_PRIORITY,
+                    &sAppTaskHandle) != pdPASS)
+    {
+        PLAT_LOG("Failed to create app task");
+        while (true)
+            ;
+    }
+    return ret;
+}
+
+int AppTask::Init()
+{
+    CHIP_ERROR ret;
+
+    cc32xxLogInit();
+
+    /* Initial Terminal, and print Application name */
+    DisplayBanner();
+
+    // Init Chip memory management before the stack
+    PLAT_LOG("Initialize Memory");
+    chip::Platform::MemoryInit();
+
+    PLAT_LOG("Initialize Wi-Fi");
+    WiFi_init();
+
+    PLAT_LOG("Initialize CHIP stack");
+    ret = PlatformMgr().InitChipStack();
+    if (ret != CHIP_NO_ERROR)
+    {
+        PLAT_LOG("PlatformMgr().InitChipStack() failed");
+        while (true)
+            ;
+    }
+
+    // Init ZCL Data Model and start server
+    PLAT_LOG("Initialize Server");
+    static chip::CommonCaseDeviceServerInitParams initParams;
+    (void) initParams.InitializeStaticResourcesBeforeServerInit();
+    chip::Server::GetInstance().Init(initParams);
+
+    // Initialize device attestation config
+    PLAT_LOG("Initialize device attestation config");
+#ifdef CC32XX_ATTESTATION_CREDENTIALS
+    SetDeviceAttestationCredentialsProvider(CC32XX::GetCC32XXDacProvider());
+#else
+
+    SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
+#endif
+
+    ConfigurationMgr().LogDeviceConfig();
+
+    // QR code will be used with CHIP Tool
+    PLAT_LOG("Print Onboarding Codes");
+    PrintOnboardingCodes(chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kOnNetwork));
+
+    PLAT_LOG("Start CHIPDeviceManager and Start Event Loop Task");
+    CHIPDeviceManager & deviceMgr = CHIPDeviceManager::GetInstance();
+    ret                           = deviceMgr.Init(&EchoCallbacks);
+    if (ret != CHIP_NO_ERROR)
+    {
+        PLAT_LOG("CHIPDeviceManager::Init() failed: %s", ErrorStr(ret));
+        while (1)
+            ;
+    }
+
+    AirQualitySensorManager::InitInstance(EndpointId(AIR_QUALITY_SENSOR_ENDPOINT));
```

Comment:
But the Matter thread is running by now, no?  The init here is going to race with that thread, likely causing memory corruption and whatnot....  This needs to either happen before the Matter thread spins up, or from the Matter thread, or synchronized via the Matter lock, I would think.